dbt-tests-adapter/dbt/tests/adapter/basic/test_get_catalog_for_single_relation.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+
+from dbt.adapters.capability import Capability
+from dbt_common.contracts.metadata import (
+    TableMetadata,
+    StatsItem,
+    CatalogTable,
+    ColumnMetadata,
+)
+from dbt.tests.util import run_dbt, get_connection
+
+models__my_model_sql = """
+{{
+    config(
+        materialized='view',
+    )
+}}
+
+select * from {{ ref('my_seed') }}
+"""
+
+seed__my_seed_csv = """id,first_name,email,ip_address,updated_at
+1,Larry,lking0@miitbeian.gov.cn,69.135.206.194,2008-09-12 19:08:31
+"""
+
+
+class BaseGetCatalogForSingleRelation:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"name": "get_catalog_for_single_relation"}
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "my_seed.csv": seed__my_seed_csv,
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": models__my_model_sql,
+        }
+
+    @pytest.fixture(autouse=True)
+    def clean_up(self, project):
+        yield
+        with project.adapter.connection_named("__test"):
+            relation = project.adapter.Relation.create(
+                database=project.database, schema=project.test_schema
+            )
+            project.adapter.drop_schema(relation)
+
+    pass
+
+    def test_get_catalog_for_single_relation(self, project):
+        results = run_dbt(["seed"])
+        assert len(results) == 1
+
+        results = run_dbt(["run"])
+        assert len(results) == 1
+
+        if project.adapter.supports(Capability.GetCatalogForSingleRelation):
```

Comment:
I don't think I would account for this here. If the adapter does not support this macro, then the maintainer will probably just not implement the test.